server/lyft/decorators/audit_project_commands_wrapper.go

```diff
@@ -0,0 +1,140 @@
+package decorators
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/events"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/lyft/aws/sns"
+)
+
+// AuditProjectCommandWrapper is a decorator that notifies sns topic
+// about the state of the command. It is used for auditing purposes
+type AuditProjectCommandWrapper struct {
+	SnsWriter sns.Writer
+	events.ProjectCommandRunner
+}
+
+func (p *AuditProjectCommandWrapper) Apply(ctx models.ProjectCommandContext) models.ProjectResult {
+	id := uuid.New()
+	startTime := time.Now()
+
+	applyEvent := &ApplyEvent{
```

Comment:
Also can you add revision? I missed it in the spec but i added it here:
https://github.com/lyft/deployapi/blob/master/pkg/postgres/migrations/1_initial_schema.up.sql#L51

Reply:
Revision is in the schema :) I'm using `ctx.Pull.HeadCommit` as a value for the field.